src/vs/editor/common/viewModel/monospaceLineBreaksComputer.ts

```diff
@@ -353,13 +357,17 @@ function createLineBreaksFromPreviousLineBreaks(classifier: WrappingCharacterCla
 	return previousBreakingData;
 }
 
-function createLineBreaks(classifier: WrappingCharacterClassifier, lineText: string, tabSize: number, firstLineBreakColumn: number, columnsForFullWidthChar: number, wrappingIndent: WrappingIndent): LineBreakData | null {
+function createLineBreaks(classifier: WrappingCharacterClassifier, _lineText: string, injectedTexts: LineInjectedText[] | null, tabSize: number, firstLineBreakColumn: number, columnsForFullWidthChar: number, wrappingIndent: WrappingIndent): LineBreakData | null {
+	const lineText = LineInjectedText.applyInjectedText(_lineText, injectedTexts);
+
 	if (firstLineBreakColumn === -1) {
+		// TODO: return LineBreakData that only has injected text if necessary
 		return null;
 	}
 
 	const len = lineText.length;
 	if (len <= 1) {
+		// TODO: return LineBreakData that only has injected text if necessary
```

Comment:
We need to do something here. This `if` statement can be hit by an empty line on which a 1 character injected text was injected.

Similarly, we should not let this `if` statement fall through. i.e. the algorithm below is not prepared for `len` to be `<= 1` (it might go in an endless loop).